extensions/amp-analytics/0.1/events.js

```diff
@@ -409,6 +411,84 @@ export class IniLoadTracker extends EventTracker {
 
 
 /**
+ * Tracks timer events.
+ */
+export class TimerEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+    /** @const @private {!Array<number>} */
+    this.trackers_ = [];
+  }
+
+  /** @override */
+  dispose() {
+    const win = this.root.ampdoc.win;
+    this.trackers_.forEach(intervalId => {
+      win.clearInterval(intervalId);
+    });
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    const timerSpec = config['timerSpec'];
+    user().assert(timerSpec && typeof timerSpec == 'object',
+        'Bad timer specification');
+    user().assert('interval' in timerSpec,
+        'Timer interval specification required');
+    const interval = Number(timerSpec['interval']) || 0;
+    user().assert(interval >= MIN_TIMER_INTERVAL_SECONDS,
+        'Bad timer interval specification');
+    const maxTimerLength = 'maxTimerLength' in timerSpec ?
+        Number(timerSpec['maxTimerLength']) : DEFAULT_MAX_TIMER_LENGTH_SECONDS;
+    user().assert(maxTimerLength == null || maxTimerLength > 0,
+        'Bad maxTimerLength specification');
+    const callImmediate = 'immediate' in timerSpec ?
+        Boolean(timerSpec['immediate']) : true;
```

Comment:
I think this is kind of risky. we never check that the `timerSpec['immediate']` is a boolean type. 
`Boolean('false') will not throw user error, but may create unexpected behavior. But since this is the behavior from old code. I am OK with it.

Reply:
I hate it as well from the old code, but was afraid to change since it'd be a breaking change, unlike the numerics code above.